packages/@aws-cdk/aws-lambda/lib/image-function.ts

```diff
@@ -41,8 +43,14 @@ export abstract class DockerImageCode {
    */
   public static fromImageAsset(directory: string, props: AssetImageCodeProps = {}): DockerImageCode {
     return {
-      _bind() {
-        return new AssetImageCode(directory, props);
+      _bind(arch?: Architecture) {
+        return new AssetImageCode(directory, {
+          /**
+           * determine the platform from `architecture`.
+           */
+          platform: arch?.dockerPlatform ? DockerPlatform.custom(arch.dockerPlatform) : undefined,
```

Comment:
Suggestion: align the style like this
https://github.com/aws/aws-cdk/blob/f52256ff49ea1699452abbd3c70cf1d4526fa80a/packages/cdk-assets/lib/private/docker.ts#L57

```suggestion
          ...arch?.dockerPlatform ? { platform: DockerPlatform.custom(arch.dockerPlatform) } : {},
```